### PR TITLE
Disconnect Flow: Submit Survey on Disconnect

### PR DESCRIPTION
**Pull request by ockham**
Use information gathered from survey steps to submit the 'Marketing Survey', which are passed as query args to the confirmation page.

This re-uses code from and is modelled after the other 'Marketing Survey' that you're asked to complete when you remove a plan via `/plans/my-plan/:site` > 'Manage Purchases' > 'Remove (plan name)'. For the required survey object layout, see https://github.com/Automattic/wp-calypso/blob/2fae5c0f96db3a30e9d2ead2efa90652d76b3aaa/client/me/purchases/remove-purchase/index.jsx#L167-L178

Since the Disconnect Flow survey has different steps (most notably no `next-adventure` and `what-better` questions), we aren't populating all fields here.

Note that this PR also normalizes the reasons. I picked names like `missing-feature`, `too-difficult`, and `too-expensive` for the routes and components since those seemed more semantic, but the marketing survey has already established `didNotInclude`, `tooHard`, and `onlyNeedFree` instead, so I have to  map to those names.

To test: You might want to use the following patch so you don't actually have to disconnect and reconnect every time:

```diff
diff --git a/client/blocks/disconnect-jetpack/index.jsx b/client/blocks/disconnect-jetpack/index.jsx
index 242e8f3..5191ecd 100644
--- a/client/blocks/disconnect-jetpack/index.jsx
+++ b/client/blocks/disconnect-jetpack/index.jsx
@@ -177,7 +177,7 @@ class DisconnectJetpack extends PureComponent {
                        }
                );
 
-               disconnectSite( siteId ).then(
+               /*disconnectSite( siteId ).then(
                        () => {
                                // Removing the domain from a domain-only site results
                                // in the site being deleted entirely. We need to call
@@ -198,7 +198,7 @@ class DisconnectJetpack extends PureComponent {
                                );
                                recordGAEvent( 'Jetpack', 'Failed Disconnected Site' );
                        }
-               );
+               );*/
        };
 
        render() {
diff --git a/client/my-sites/site-settings/disconnect-site/confirm.jsx b/client/my-sites/site-settings/disconnect-site/confirm.jsx
index 7609acd..1fbb038 100644
--- a/client/my-sites/site-settings/disconnect-site/confirm.jsx
+++ b/client/my-sites/site-settings/disconnect-site/confirm.jsx
@@ -49,11 +49,13 @@ class ConfirmDisconnection extends PureComponent {
                        },
                };
 
-               submitSurvey(
+               console.log( enrichedSurveyData( surveyData, moment(), site, purchase ) );
+
+               /*submitSurvey(
                        'calypso-disconnect-jetpack',
                        siteId,
                        enrichedSurveyData( surveyData, moment(), site, purchase )
-               );
+               );*/
        };
 
        render() {

```

* Start at `/settings/disconnect-site/:site`
* Follow thru the flow -- be sure to cover each option once. 
* At the confirmation step, click the red 'Disconnect' button.
* Verify in the dev console that the survey data that would be submitted match what you specified in the survey.
* Make sure to test at least once _without_ the above patch; since `submitSurvey` isn't a Redux but a Flux action, watch the Network tab to make sure your data are actually sent.

cc @bubel for review of the data we're sending

### CODE REVIEW

**Comment by seear** [comment]
Note: even with the test patch the site ends up disconnected for me. Probably would need a tweak in [`<DisconnectJetpack />`](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/disconnect-jetpack/index.jsx) to prevent that.

**Comment by ockham** [comment]
> Note: even with the test patch the site ends up disconnected for me. Probably would need a tweak in <DisconnectJetpack /> to prevent that.

D'oh, you're right. A `preventDefault` might help...

**Comment by ockham** [comment]
@seear: Updated the patch!

File: client/my-sites/site-settings/disconnect-site/missing-feature.jsx [hunk 1]
```diff
@@ -60,7 +60,10 @@ class MissingFeature extends PureComponent {
 					<Button
 						disabled={ isEmpty( this.state.tokens ) }
 						href={ addQueryArgs(
-							{ 'missing-features': this.state.tokens.map( this.normalizeToken ) },
+							{
+								reason: 'missing-feature',
+								text: this.state.tokens.map( this.normalizeToken ),
```
Comment (seear) [inline client/my-sites/site-settings/disconnect-site/missing-feature.jsx]: Maybe some value in sorting the tokens here so identical sets of reasons always match up in the data?

**Comment by seear** [comment]
Little glitch here, probably fine to fix in separate PR since it seems to be part of `TokenField`. The _Submit_ button is blue here but if you click it nothing happens. You have to click it again. Sort of thing that I've noticed can (rightly) stump not-so-technical people for a bit.

screen shot 2017-10-23 at 16 42 09

**Comment by ockham** [comment]
Thanks for reviewing @seear and @bubel!

> Little glitch here, probably fine to fix in separate PR since it seems to be part of `TokenField`. The _Submit_ button is blue here but if you click it nothing happens. You have to click it again. Sort of thing that I've noticed can (rightly) stump not-so-technical people for a bit.

Good spot! Will fix in a subsequent PR.

**Pull request merged**